core/shared/src/main/scala/zio/FiberFailure.scala

```diff
@@ -26,11 +26,14 @@ import zio.stacktracer.TracingImplicits.disableAutoTrace
  * This class is used to wrap ZIO failures into something that can be thrown, to
  * better integrate with Scala exception handling.
  */
-final case class FiberFailure(cause: Cause[Any]) extends Throwable(null, null, true, false) {
+final case class FiberFailure(cause: Cause[Any], javaStackTrace: Array[StackTraceElement])
```

Comment:
Added `javaStackTrace` parameter to `FiberFailure` to capture the Java stack trace at the time of failure.